sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -129,23 +129,39 @@ object UnresolvedAttribute {
     nameParts.toSeq
   }
 }
+trait UnresolvedFunctionLike extends Expression {
```

Comment:
I'd remove this trait since it is only used twice. The overhead of having an abstraction doesn't seem worth it.